### ISSUE TITLE
Pro Edition Call-To-Action Proposal

### ISSUE THREAD

**Issue by putyourlightson** [craftcms/cms#3697]
With the introduction of plugin editions in Craft 3.1, a standardised way of displaying the availability of more features in the Lite/Plus editions is being proposed.

In a discussion with some other plugin developers today, we discussed possibilities for indicating the availability of Pro features in Lite editions of plugins. Commerce Lite currently hides any pages that are Pro features (Settings → Shipping; Settings → Taxes) and the only indicators of more features are the badge on the plugin settings page and the plugin page in the plugin store.

<img width="612" alt="screenshot 2019-01-23 at 20 48 10" src="https://user-images.githubusercontent.com/2318222/51632711-419a8080-1f50-11e9-848f-68b1875ba074.png">

Some ideas we discussed are as follows. The intention is to be as unobtrusive as possible, while making it obvious that there are more goodies to be had.

## 1. The Badge

The badge is displayed in places where pages with Pro features exist. The links could be either disabled completely, or enabled and link to placeholder pages. The mark-up for these badges already exists (in Utilities and Updates).

<img width="208" alt="screenshot 2019-01-23 at 20 31 50" src="https://user-images.githubusercontent.com/2318222/51632616-0ef08800-1f50-11e9-8272-896c1409cb9b.png">

<img width="211" alt="screenshot 2019-01-23 at 20 44 29" src="https://user-images.githubusercontent.com/2318222/51632628-144dd280-1f50-11e9-8470-af841ca65167.png">

## 2. The Placeholder Page

The placeholder page is a page that would be available in the Pro edition of the plugin. In a Lite edition the user could still visit those pages but would see a placeholder image of what the page would offer in the Pro edition. This image could take the form of a silhouette or of sample output, along with a call-to-action.

![imageedit_2_5677200216](https://user-images.githubusercontent.com/2318222/51633237-81159c80-1f51-11e9-9d2a-165739dc1665.png)

---

Perhaps there is a better place to post this question, but it would be great to get the opinion of developers as well as end-users of plugins that are currently offering or considering editions. Specifically, would you consider this a useful feature or more of a naggy feature?

**Comment by khalwat** [comment]
This came up in our discussion, and I think the main reason for _making available_ some kind of standardization is just to give the user a good experience.

Plugins that may be free, but have a paid tier... or may be paid, or have another paid tier need:

- Some way to indicate to the user (without being obnoxious) that an additional paid tier exists
- Some way to indicate to the user that there is additional functionality that they could be taking advantage of
- Some way for them to click a CTA or link to upgrade to said functionality

This would solve the issues of discoverability & CTA that probably all plugin developers will encounter (including P&T with Commerce) in a standardized way that provides a good experience for the end-user.

**Comment by brandonkelly** [comment]
We’re not adamantly opposed to it, but I’m not ready to establish a specific recommendation yet. I think plugins should play around with ideas first and we’ll see if a convention emerges from that.

**Comment by putyourlightson** [comment]
Fair enough. In an ideal world, Craft developers will help guide us as to whether this is something they even want us to add to plugins and how they would like to see it implemented.

**Comment by khalwat** [comment]
Only tangentially related, but I wish `lite` was always `free`, and the base paid tier was `plus`. Just so when people see `lite`, they know what it means.

Right now, `lite` can be paid, or it can be free.

**Comment by brandonkelly** [comment]
@khalwat Interesting. We had a (very) long internal debate about edition names. We had sort of the opposite thought though – that Lite seems like it implies cheap (but not free), and that there should be some other tier below it reserved for free editions. But in the end we decided against that.

**Comment by narration-sd** [comment]
A moment interruption lets me just agree with @brandonkelly and team here.

I suspect it's very nice (and commercially effective) to be able to offer a 'lite' version, without 'pro' features as the telling point.

And especially, it seems free should be: in the case when that word is used....

For commercial, it's another differentiator...

**Comment by lindseydiloreto** [comment]
@khalwat @brandonkelly "Lite" means cheap to me, "Free" means free.

That said, plugins should be able to have a "Free" edition.

**Comment by brandonkelly** [comment]
That’s how we felt too @lindseydiloreto, but can’t stomach calling an edition “Free”.

**Comment by narration-sd** [comment]
Ah, looks like I misinterpreted you, @brandonkelly. And yes, 'free' has many effects on people.

Even a good thesaurus didn't help: introductory, unrestricted, priceless(!)...

hmm

**Comment by carlcs** [comment]
Solo

**Comment by carlcs** [comment]
And a 👎 for the idea to advertise Pro editions in the sidebar and client facing up in general. What’s next? Sponsored by Xy banners?

**Comment by putyourlightson** [comment]
Thanks @carlcs, noted.

**Comment by narration-sd** [comment]
I had a waking thought this morning about the naming problem. I suspect the solution lies in separating concepts: getting pricing thoughts out of the names. Then price, is just price, no, and listed separately?

So we could have Stage 1, Stage 2, etc. for capability levels. And this means also that Stage 2 or N could also be free, if that's the provider's choice. Or could get free, or get a price, in some future. Flexibility.

Alternatives for 'Stage'?  This is where Carl's slant with 'Solo' gets important, as you want something friendly instead of project management words.

It might be good just to let the provider provide the names. As in 'Basic' or  'Base' or 'Introductory' etc.. Which also would let the clever do something like Tokyo Dawn did for their non-free versions of audio plugins, call those the 'Gentleman's Edition'...

To be helpful and guiding, Craft Store might propose a name which can be edited. 

And then a neutral possibility for the default shows possibility:  'Edition' . Thus, Edition 1, 2, etc., which could easily become  Edition A, Edition B, Entry Edition, Édition superlatif, etc..

**Comment by putyourlightson** [comment]
Thanks for the input @narration-sd. Pixel & Tonic have made it clear that the naming convention for editions is fixed, and as I previously stated, I'd prefer that this thread not become a discussion about that, but that it stay focused on the original question. @carlcs's opinion was quite clear and I'd welcome any other opinions on the proposed ideas.

**Comment by narration-sd** [comment]
Well, in my own thoughts I probably missed that, Ben. A little less formal than some I admit they are ;) -- privilege of elderness...

Anyway, I think what I said won't hurt, as the future is always becoming, isn't it, and not less certainly with Craft and its persons...

Also, did quite appreciate what @carlcs Carl was getting at, to be  sure
